lab_4_retrieval_w_clustering/main.py

```diff
@@ -380,6 +616,25 @@ def retrieve_relevant_documents(self, query: str, n_neighbours: int) -> list[tup
             list[tuple[float, str]]: Relevant documents with their distances.
         """
 
+        if not query or not isinstance(query, str):
+            raise ValueError("Oops! Invalid input")
+        if not n_neighbours or not isinstance(n_neighbours, int) or n_neighbours <= 0:
+            raise ValueError("Oops! Invalid input")
+        query_token = self._db.get_tokenizer().tokenize(query)
+        if query_token is None:
+            raise ValueError("Oops! It's empty")
```

Comment:
yes, it seems original, but for user it is useless input, do you understand?

Reply:
output message*

Reply:
I guess so, fixed